### PR TITLE
Update Catppuccin for all variants

### LINKED ISSUE #686: Other Catppuccin variants
**What problem does this feature solve?**

It spreads the rest of the beauty of catppuccin.

**What feature do you propose?**

Would you be open to a follow-up PR that adds the other three variants? 
#685 is `mocha`, the other three would be `frappe`, `macchiato`, and `latte`.

### PR DESCRIPTION
**Pull request by jolheiser**
Resolves #686 

❗ I've removed the `catppuccin` theme and replaced it with `catppuccin-mocha` in order to make them all standard. Let me know if you want me to change it back to just `catppuccin`, I realize it's potentially a breaking change but there hasn't been a new release cut and it's only been merged for a few days, so I changed it for now.

The following previews are HTML of the HTML code of these variants, because inception is cool and the Go lexer doesn't show off enough of the colors based on just the style's Go file. 😅 

## Latte
latte

## Frappe
frappe

## Macchiato
macchiato

## Mocha
mocha

### CODE REVIEW

File: styles/catppuccin-frappe.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// CatppuccinFrappe a soothing low-saturation, low-contrast dark pastel theme for the high-spirited
+var CatppuccinFrappe = Register(chroma.MustNewStyle("catppuccin-frappe", chroma.StyleEntries{
+	chroma.TextWhitespace:        ctpFrappeSurface0,
+	chroma.Comment:               "italic " + ctpFrappeSurface2,
+	chroma.CommentPreproc:        ctpFrappeBlue,
+	chroma.Keyword:               ctpFrappeMauve,
+	chroma.KeywordPseudo:         "bold " + ctpFrappeMauve,
+	chroma.KeywordType:           ctpFrappeFlamingo,
```
Comment (nekowinston) [inline styles/catppuccin-frappe.go:34]:
```suggestion
	chroma.KeywordType:           ctpFrappeYellow
```

File: styles/catppuccin-latte.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// CatppuccinLatte a soothing light pastel theme for the high-spirited
+var CatppuccinLatte = Register(chroma.MustNewStyle("catppuccin-latte", chroma.StyleEntries{
+	chroma.TextWhitespace:        ctpLatteSurface0,
+	chroma.Comment:               "italic " + ctpLatteSurface2,
+	chroma.CommentPreproc:        ctpLatteBlue,
+	chroma.Keyword:               ctpLatteMauve,
+	chroma.KeywordPseudo:         "bold " + ctpLatteMauve,
+	chroma.KeywordType:           ctpLatteFlamingo,
```
Comment (nekowinston) [inline styles/catppuccin-latte.go:34]:
```suggestion
	chroma.KeywordType:           ctpFrappeYellow
```

File: styles/catppuccin-macchiato.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// CatppuccinMacchiato a soothing mid-saturation, mid-contrast dark pastel theme for the high-spirited
+var CatppuccinMacchiato = Register(chroma.MustNewStyle("catppuccin-macchiato", chroma.StyleEntries{
+	chroma.TextWhitespace:        ctpMacchiatoSurface0,
+	chroma.Comment:               "italic " + ctpMacchiatoSurface2,
+	chroma.CommentPreproc:        ctpMacchiatoBlue,
+	chroma.Keyword:               ctpMacchiatoMauve,
+	chroma.KeywordPseudo:         "bold " + ctpMacchiatoMauve,
+	chroma.KeywordType:           ctpMacchiatoFlamingo,
```
Comment (nekowinston) [inline styles/catppuccin-macchiato.go:34]:
```suggestion
	chroma.KeywordType:           ctpFrappeYellow
```

File: styles/catppuccin-mocha.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+)
+
+// CatppuccinMocha a soothing high-saturation, high-contrast dark pastel theme for the high-spirited
+var CatppuccinMocha = Register(chroma.MustNewStyle("catppuccin-mocha", chroma.StyleEntries{
+	chroma.TextWhitespace:        ctpMochaSurface0,
+	chroma.Comment:               "italic " + ctpMochaSurface2,
+	chroma.CommentPreproc:        ctpMochaBlue,
+	chroma.Keyword:               ctpMochaMauve,
+	chroma.KeywordPseudo:         "bold " + ctpMochaMauve,
+	chroma.KeywordType:           ctpMochaFlamingo,
```
Comment (nekowinston) [inline styles/catppuccin-mocha.go:34]:
```suggestion
	chroma.KeywordType:           ctpFrappeYellow
```

File: styles/catppuccin-latte.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.KeywordConstant:       "italic " + ctpLatteMauve,
+	chroma.Operator:              ctpLatteSky,
+	chroma.OperatorWord:          "bold " + ctpLatteSky,
+	chroma.Name:                  ctpLatteLavender,
+	chroma.NameBuiltin:           "italic " + ctpLatteText,
```
Comment (nekowinston) [inline styles/catppuccin-latte.go:39]:
```suggestion
	chroma.NameBuiltin:           "italic " + ctpMochaPeach,
```

File: styles/catppuccin-mocha.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.KeywordConstant:       "italic " + ctpMochaMauve,
+	chroma.Operator:              ctpMochaSky,
+	chroma.OperatorWord:          "bold " + ctpMochaSky,
+	chroma.Name:                  ctpMochaLavender,
+	chroma.NameBuiltin:           "italic " + ctpMochaText,
```
Comment (nekowinston) [inline styles/catppuccin-mocha.go:39]:
```suggestion
	chroma.NameBuiltin:           "italic " + ctpMochaPeach,
```

File: styles/catppuccin-frappe.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.NameOther:             ctpFrappePeach,
+	chroma.Punctuation:           ctpFrappeText,
+	chroma.LiteralString:         ctpFrappeGreen,
+	chroma.LiteralStringDoc:      ctpFrappeGreen,
+	chroma.LiteralStringInterpol: ctpFrappeGreen,
+	chroma.LiteralStringEscape:   ctpFrappeBlue,
+	chroma.LiteralStringRegex:    ctpFrappeBlue,
+	chroma.LiteralStringSymbol:   ctpFrappeGreen,
+	chroma.LiteralStringOther:    ctpFrappeGreen,
+	chroma.LiteralNumber:         ctpFrappeTeal,
```
Comment (nekowinston) [inline styles/catppuccin-frappe.go:60]:
```suggestion
	chroma.LiteralNumber:         ctpMochaPeach,
```

File: styles/catppuccin-latte.go [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.NameOther:             ctpLattePeach,
+	chroma.Punctuation:           ctpLatteText,
+	chroma.LiteralString:         ctpLatteGreen,
+	chroma.LiteralStringDoc:      ctpLatteGreen,
+	chroma.LiteralStringInterpol: ctpLatteGreen,
+	chroma.LiteralStringEscape:   ctpLatteBlue,
+	chroma.LiteralStringRegex:    ctpLatteBlue,
+	chroma.LiteralStringSymbol:   ctpLatteGreen,
+	chroma.LiteralStringOther:    ctpLatteGreen,
+	chroma.LiteralNumber:         ctpLatteTeal,
```
Comment (nekowinston) [inline styles/catppuccin-latte.go:60]:
```suggestion
	chroma.LiteralNumber:         ctpMochaPeach,
```

File: styles/catppuccin-macchiato.go [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.NameOther:             ctpMacchiatoPeach,
+	chroma.Punctuation:           ctpMacchiatoText,
+	chroma.LiteralString:         ctpMacchiatoGreen,
+	chroma.LiteralStringDoc:      ctpMacchiatoGreen,
+	chroma.LiteralStringInterpol: ctpMacchiatoGreen,
+	chroma.LiteralStringEscape:   ctpMacchiatoBlue,
+	chroma.LiteralStringRegex:    ctpMacchiatoBlue,
+	chroma.LiteralStringSymbol:   ctpMacchiatoGreen,
+	chroma.LiteralStringOther:    ctpMacchiatoGreen,
+	chroma.LiteralNumber:         ctpMacchiatoTeal,
```
Comment (nekowinston) [inline styles/catppuccin-macchiato.go:60]:
```suggestion
	chroma.LiteralNumber:         ctpMochaPeach,
```

File: styles/catppuccin-mocha.go [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+	chroma.NameOther:             ctpMochaPeach,
+	chroma.Punctuation:           ctpMochaText,
+	chroma.LiteralString:         ctpMochaGreen,
+	chroma.LiteralStringDoc:      ctpMochaGreen,
+	chroma.LiteralStringInterpol: ctpMochaGreen,
+	chroma.LiteralStringEscape:   ctpMochaBlue,
+	chroma.LiteralStringRegex:    ctpMochaBlue,
+	chroma.LiteralStringSymbol:   ctpMochaGreen,
+	chroma.LiteralStringOther:    ctpMochaGreen,
+	chroma.LiteralNumber:         ctpMochaTeal,
```
Comment (nekowinston) [inline styles/catppuccin-mocha.go:60]:
```suggestion
	chroma.LiteralNumber:         ctpMochaPeach,
```

**Review by nekowinston — COMMENTED** [review]
Those are just the things that immediately jumped out to me.
I'd either like to spend some more time tomorrow reviewing the highlights, or send another PR to implement those changes, should I find any.

**Comment by jolheiser** [comment]
@nekowinston did you want to do another review or should I appease the linter gods?

**Comment by alecthomas** [comment]
I don't suppose either of you have encountered a commonly used machine-readable format for themes?

**Comment by jolheiser** [comment]
> I don't suppose either of you have encountered a commonly used machine-readable format for themes?

I haven't, because most have their own set of keys/values available.
Most of what I've seen is in JSON, but I've seen XML used here and there.

**Comment by sgoudham** [comment]
Looks 🔥 🔥 Excited to see this get merged!

**Comment by alecthomas** [comment]
I think we're waiting for another pass by @nekowinston , let's give it another day or two then remind me to merge it.

**Comment by jolheiser** [comment]
@alecthomas I think we're okay to merge this. 🚀 
If @catppuccin requires further changes we can address a follow-up PR. At least then it will only be color changes, the names should remain stable.

**Review by nekowinston — APPROVED** [review]
Sorry for causing a delay here - this seems fine. We should really work on a more defined styleguide, so that we have more defined docs which colors should map to which tokens; but this seems good for now.

**Comment by alecthomas** [comment]
👍

Thanks everyone!

**Pull request merged**